Source/WebCore/dom/ElementInlines.h

```diff
@@ -212,4 +212,12 @@ inline bool isInTopLayerOrBackdrop(const RenderStyle& style, const Element* elem
     return (element && element->isInTopLayer()) || style.styleType() == PseudoId::Backdrop;
 }
 
+inline void Element::hideNonce()
+{
+    // In the common case, Elements don't have a nonce parameter to hide.
+    if (LIKELY(!isConnected() || !hasAttributeWithoutSynchronization(HTMLNames::nonceAttr)))
```

Comment:
Should we consider setting a bit flag on ElementData or EventTarget indicating whether we have a nonceAttr or not? Otherwise, this is O(k) search where k is the number of attributes, right?

Reply:
I can try that and see if it helps the Speedometer score. If it doesn't have a meaningful impact on the Speedometer score then it is not worth a bit flag?

Reply:
These things add up though. It might be too small to measure on their own but 3-4 of these can add up to a cost high.

Reply:
That's very true and the reason why I am writing these micro optimizations in the first place. I'll look into it.